sample_apps/run_example.sh

```diff
@@ -283,19 +293,46 @@ This_fn=""
 # Wrapper to run a command w/ parameters.
 # ###########################################################################
 function run_cmd() {
-   echo
+   echo " "
    This_fn="${FUNCNAME[1]}"
    if [ "${Prev_fn}" != "${This_fn}" ]; then
        echo "******************************************************************************"
        echo "${Me}: Running ${FUNCNAME[1]} "
-       echo
        Prev_fn="${This_fn}"
    fi
-   set -x
+   # Only execute the commands if --dry-run is OFF
+   if [ ${DryRun} -eq 0 ]; then
+        set -x
 
-   "$@"
+        "$@"
 
-   set +x
+        set +x
+    else
+        # Print what would have been executed, so user can learn from this.
+        echo "$@"
+    fi
+}
+
+# ###########################################################################
+# Wrappers around pushd / popd to also honor --dry-run mode, so user can
+# inspect changes to dirs before running various commands.
+# ###########################################################################
+function run_pushd() {
+    local todir="$1"
+    if [ ${DryRun} -eq 0 ]; then
+        pushd "${todir}" > /dev/null 2>&1
+    else
+        echo "pushd ${todir}"
+    fi
+}
+
+function run_popd() {
```

Comment:
Added this  and `run_pushd` above to work correctly if `--dry-run` is ON or OFF.